python/pyarrow/table.pxi

```diff
@@ -1450,8 +1450,76 @@ cdef _sanitize_arrays(arrays, names, schema, metadata,
             converted_arrays.append(item)
     return converted_arrays
 
+cdef class _Table(_PandasConvertible):
+    """Internal: An interface for common table operations."""
 
-cdef class RecordBatch(_PandasConvertible):
+    def __init__(self):
+        raise TypeError("This object is not instantiable, "
+                        "use a subclass instead.")
+
+    def drop_null(self):
+        """
+        Remove missing values from a Table.
```

Comment:
I think that's fine to just use a single example. We could always add a comment like `# or pa.RecordBatch.from_pandas(df)` above the equivalent Table line, to make it clear how to run the equivalent example with a RecordBatch instead of a Table

Reply:
Or adding a standard sentence like "The following example uses a Table, but it works the same for RecordBatch". Or would that just be unnecessary noise in most cases?